src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -140,11 +152,32 @@ public bool ExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionConte
                 }
 
                 progressTracker.ItemCompleted();
-                return;
             }
+            else
+            {
+                Format(args.TextView, document, selectionOpt: null, cancellationToken);
+            }
+        }
 
-            ShowGoldBarForCodeCleanupConfigurationIfNeeded(document);
-            Format(args.TextView, document, selectionOpt: null, cancellationToken);
+        private static void ABTest(Document document, Workspace workspace)
+        {
+            // AB Test
+            // If the feature is OFF and the feature options have not yet been set to their assigned flight, do so now
+            if (!workspace.Options.GetOption(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting, document.Project.Language)
+                && !workspace.Options.GetOption(CodeCleanupABTestOptions.ParticipatedInExperiment))
+            {
+                var experimentationService = document.Project.Solution.Workspace.Services.GetService<IExperimentationService>();
+
+                if (experimentationService != null)
+                {
+                    if (experimentationService.IsExperimentEnabled(s_experimentName))
+                    {
+                        workspace.Options = workspace.Options.WithChangedOption(new OptionKey(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting, document.Project.Language), true);
+                    }
+
+                    workspace.Options = workspace.Options.WithChangedOption(new OptionKey(CodeCleanupABTestOptions.ParticipatedInExperiment), true);
```

Comment:
This is being set _outside_ the IsExperimentEnabled call. So if we ship this (but don't yet move anybody to B), won't this set ParticipatedInExperiment = true, and then never let us move them to B?